src/Storage/Telemetry/TelemetryEnrichingMiddleware.cs

```diff
@@ -0,0 +1,149 @@
+    {
+        var actions = new Dictionary<string, Action<Claim, Activity>>(StringComparer.OrdinalIgnoreCase)
+        {
+            {
+                AltinnCoreClaimTypes.UserId,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.id", claim.Value);
+                }
+            },
+            {
+                AltinnCoreClaimTypes.PartyID,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.party.id", claim.Value);
+                }
+            },
+            {
+                AltinnCoreClaimTypes.AuthenticationLevel,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.authentication.level", claim.Value);
+                }
+            },
+            {
+                AltinnCoreClaimTypes.Org,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.application.owner.id", claim.Value);
+                }
+            },
+            {
+                AltinnCoreClaimTypes.OrgNumber,
+                static (claim, activity) =>
+                {
+                    activity.SetTag("user.organization.number", claim.Value);
+                }
+            },
+            {
+                "authorization_details",
+                static (claim, activity) =>
+                {
+                    SystemUserClaim claimValue = JsonSerializer.Deserialize<SystemUserClaim>(claim.Value);
+                    activity.SetTag("user.system.id", claimValue?.Systemuser_id[0] ?? null);
+                    activity.SetTag("user.system.owner.number", claimValue?.Systemuser_org.ID ?? null);
+                }
+            },
+        };
+
+        return actions.ToFrozenDictionary();
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TelemetryEnrichingMiddleware"/> class.
+    /// </summary>
+    /// <param name="next">The next middleware in the pipeline.</param>
+    /// <param name="logger">The logger instance.</param>
+    /// <param name="generalSettings">Configuration object used to hold general settings for the storage application.</param>
+    public TelemetryEnrichingMiddleware(RequestDelegate next, ILogger<TelemetryEnrichingMiddleware> logger, IOptions<GeneralSettings> generalSettings)
+    {
+        _next = next;
+        _logger = logger;
+        _disableTelemetryForMigration = generalSettings.Value.DisableTelemetryForMigration;
+    }
+
+    /// <summary>
+    /// Invokes the middleware to process the HTTP context.
+    /// </summary>
+    /// <param name="context">The HTTP context.</param>
+    public async Task InvokeAsync(HttpContext context)
+    {
+        var activity = context.Features.Get<IHttpActivityFeature>()?.Activity;
+        if (activity is null || !context.Request.Path.ToString().Contains("storage/api/"))
```

Comment:
Unsure if this is sufficient, as the original also checked for another condition
https://github.com/Altinn/altinn-storage/pull/638/files#diff-8ee9e0d48ee34118f74728195d86ac39a858405f7fdf0df45b6a8140377e3a5cL49

Reply:
Tried to handle this is the newest commit